src/main/java/Parser.java

```diff
@@ -0,0 +1,87 @@
+import java.util.Scanner;
+
+import java.io.IOException;
+
+public class Parser {
+
+    private Storage storage;
+    private TaskList taskList;
+    private Ui ui;
+
+    public Parser(Storage storage, TaskList taskList, Ui ui) {
+        this.storage = storage;
+        this.taskList = taskList;
+        this.ui = ui;
+    }
+
+    public void parse() {
```

Comment:
Very intuitive way of handling exceptions! I see you are going for the "for each command handle it own exceptions" approach. But I think it can get a bit cluttered the more try/catch blocks you add. Maybe I could suggest trying the whole block of commands and catching them at the end?